src/java/org/tensorics/core/booleans/lang/OngoingBooleanDetectionIterableAware.java

```diff
@@ -0,0 +1,20 @@
+/**
+ * Copyright (c) 2016 European Organisation for Nuclear Research (CERN), All Rights Reserved.
+ */
+
+package org.tensorics.core.booleans.lang;
+
+public class OngoingBooleanDetectionIterableAware {
+
+    private Iterable<Boolean> iterable;
+
+    public OngoingBooleanDetectionIterableAware(Iterable<Boolean> iterable) {
+        this.iterable = iterable;
+    }
+
+    // TODO should be index of the change? or the case is not valid at all? (only later interesting for
+    // tensors/functions)
+    public Integer changes() {
```

Comment:
Interesting question ... indices on an iterable are dangerous because an iterable is not guaranteed to be always iterate in the same order (e.g. unsorted collections).
Also, to be consistent with the tensoric use case, this would return an Iterable of all changes, not just the first.
Anyway, no idea if you're going to need this. If not, it's perhaps easier to leave it out. Otherwise, you might have to limit it e.g. to a list instead of an iterable (even though I agree this would not be consistent with the rest of the DSL).